eureka2-core/src/main/java/com/netflix/eureka2/config/BasicEurekaRegistryConfig.java

```diff
@@ -9,32 +11,36 @@
  */
 public class BasicEurekaRegistryConfig implements EurekaRegistryConfig {
 
-    private static final String EVICTION_TIMEOUT_MS = "30000";
-    private static final String EVICTION_STRATEGY_TYPE = StrategyType.PercentageDrop.name();
-    private static final String EVICTION_STRATEGY_VALUE = "20";
+    public static final String EVICTION_TIMEOUT_MS = "30000";
+    public static final String EVICTION_STRATEGY_TYPE = StrategyType.PercentageDrop.name();
+    public static final String EVICTION_STRATEGY_VALUE = "20";
 
-    private String evictionTimeoutMs = System.getProperty("eureka.registry.evictionTimeoutMs", EVICTION_TIMEOUT_MS);
-    private String evictionStrategyType = System.getProperty("eureka.registry.evictionStrategy.type", EVICTION_STRATEGY_TYPE);
-    private String evictionStrategyValue = System.getProperty("eureka.registry.evictionStrategy.value", EVICTION_STRATEGY_VALUE);
+    private Long evictionTimeoutMs = Long.parseLong(
+            System.getProperty(evictionTimeoutMsName, EVICTION_TIMEOUT_MS)
+    );
+    private StrategyType evictionStrategyType = StrategyType.valueOf(
```

Comment:
We should handle more gracefully errors here, as exceptions are very cryptic when class initialization fails. If property value cannot be parsed, we should log a warning and use a default.